### PR TITLE
George redirecting

### PR DESCRIPTION
**Pull request by Georgehe4**
Export now works differently.

### CODE REVIEW

**Comment by Georgehe4** [comment]
@devoncarew

File: lib/embed/embed.dart [hunk 1]
```diff
@@ -84,17 +84,16 @@ class PlaygroundMobile {
...
+    Uri url = Uri.parse(window.location.toString());
+    if (url.hasQuery) {
+      String id = url.queryParameters['id'];
       if (isLegalGistId(id)) {
         _showGist(id);
         return;
       }
     }
-
+
```
Comment (devoncarew) [inline lib/embed/embed.dart]: Can you change the settings on your editor to strip ws at the end of lines?

File: lib/embed/embed.dart [hunk 2]
```diff
@@ -279,7 +278,7 @@ class PlaygroundMobile {
 
   void _export() {
     window.open(
-        "/newpad?dart=${Uri.encodeQueryComponent(context.dartSource)}&html=${Uri.encodeQueryComponent(context.htmlSource)}&css=${Uri.encodeQueryComponent(context.cssSource)}",
+        "/index.html?dart=${Uri.encodeQueryComponent(context.dartSource)}&html=${Uri.encodeQueryComponent(context.htmlSource)}&css=${Uri.encodeQueryComponent(context.cssSource)}",
```
Comment (devoncarew) [inline lib/embed/embed.dart]: This is great - we don't need a new path.

File: lib/playground.dart [hunk 3]
```diff
@@ -181,11 +181,12 @@ class Playground implements GistContainer, GistController {
...
     bool loadedFromSaved = false;
-    if (querySelector('#dart-code') != null) {
+    Uri url = Uri.parse(window.location.toString());
+    if (url.hasQuery && url.queryParameters['dart'] != null) {
       Gist blankGist = createSampleGist();
-      blankGist.getFile('main.dart').content = Uri.decodeQueryComponent(querySelector('#dart-code').innerHtml);
-      blankGist.getFile('index.html').content = Uri.decodeQueryComponent(querySelector('#html-code').innerHtml);
-      blankGist.getFile('styles.css').content = Uri.decodeQueryComponent(querySelector('#css-code').innerHtml);
+      blankGist.getFile('main.dart').content = url.queryParameters['dart'];
+      blankGist.getFile('index.html').content = url.queryParameters['html'];
```
Comment (devoncarew) [inline lib/playground.dart]: Can you manually verify that not having an html or css parameter on the url will not cause issues (that we won't have problems if we're assigning `null` to `.content`)?

**Comment by devoncarew** [comment]
lgtm w/ the manual verification. I really like that the front-end server is simpler with this change.

**Comment by devoncarew** [comment]
lgtm!

**Comment by Georgehe4** [comment]
Merging
#541

**Pull request merged**